app/app.ts

```diff
@@ -434,6 +435,27 @@ function setupIpc(localSettings: LocalSettings, scrSettings: ScrSettings) {
     })
   })
 
+  ipcMain.handle('replayShieldBatteryData', async (event, replayPath) => {
```

Comment:
I don't really understand why this is split up from getting the replay header, by doing it this way you've basically guaranteed that every time we look at a replay we have to:

- Make 2 trips across the IPC boundary
- Open the file twice
- Read the file twice
- Parse the file twice

And we're basically always doing all of these at the exact same time. Why not just return the ShieldBattery data as part of the normal process of reading the replay header? The data is of the same importance to us, you basically always want it, and reading it is a negligible additional cost to parsing the replay header normally. You should probably be logging cases where the replay has SB data but it cannot be parsed (for debug purposes), but from a user perspective, this can be handled identically to the replay not having that data in the first place, so there's really no need to generate 2 promises from this operation